library/alloc/src/string.rs

```diff
@@ -1585,6 +1585,32 @@ impl String {
         let slice = self.vec.into_boxed_slice();
         unsafe { from_boxed_utf8_unchecked(slice) }
     }
+
+    /// Consumes and leaks the `String`, returning a mutable reference `&'static mut str` to the
+    /// contents.
+    ///
+    /// This function is similar to the [`leak`] function on [`Box`].
+    ///
+    /// This function is mainly useful for data that lives for the remainder of
+    /// the program's life. Dropping the returned reference will cause a memory
+    /// leak.
+    ///
+    /// [`leak`]: Box::leak
+    ///
+    /// # Examples
+    ///
+    /// Basic usage:
+    ///
+    /// ```
+    /// let my_string = String::from("HELLO!");
+    /// let static_ref: &'static mut str = my_string.leak();
+    /// static_ref.make_ascii_lowercase();
+    /// assert_eq!(static_ref, "hello!");
+    /// ```
+    #[inline]
+    pub fn leak(self) -> &'static mut str {
```

Comment:
I noticed that the Box version of this isn't `'static`, but does `fn leak<'a>(b: Box<T>) -> &'a mut T `

Should this maybe also be lifetime-generic?
```suggestion
    pub fn leak<'a>(self) -> &'a mut str {
```

Reply:
Box's version needs that because the `T` in `Box<T>` might not be `: 'static`. E.g. `Box<&'x u32>`. The leaked reference from `Box::leak` can not be `'static` in that case: it must not outlive `'x`. This is not an issue here, as `str` doesn't contain any references/lifetimes. So `&'static str` should be fine here.